### PR TITLE
Fixing long transactions

### LINKED ISSUE #21: Long transaction are not working in Desktop wallet
**Describe the bug**
Currently, in Desktop wallet only transactions with up to 8 inputs are supported, however, you can sign the long transactions using CLI.

**Environment information:**

- OS: all
- Platform: all
- Firmware Version: 1.1.0

**Actual behavior**
Support for every transaction.

**Expected behavior**
Only transactions with up to 8 inputs are working.

**Additional context**
For communicating with a desktop wallet hardware-wallet-daemon is used, which uses [Devicer](https://github.com/SkycoinProject/hardware-wallet-go/blob/master/src/skywallet/skywallet.go#L56) interface to communicate with Skywallet. There we have only TransactionSign method, but not SignTx, which supports long transactions. 

**Possible implementation**
We can solve this issue, by moving the whole state machine from the CLI to TransactionSign. In such a way, messages, that used signing for small transactions become deprecated and we don't need to change anything in Desktop wallet.

### PR DESCRIPTION
**Pull request by Midren**
Fixes #21 

Changes:
- Add long transaction signing, as default
- Make the output of signing long transaction the same, as was for small

Does this change need to mentioned in CHANGELOG.md?
no

Requires changes in protobuff specs?
no

Requires testing
yes

Comments about testing , should you have some
Test signing long transaction from hardware-wallet-daemon

### CODE REVIEW

File: src/skywallet/transaction.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+						msg, err = s.sendInputs(index, len(s.Inputs)-index)
+						if err != nil {
+							return nil, err
+						}
+						s.state++
+						index = 0
+					}
+					index += 8
+				} else {
+					return nil, fmt.Errorf("protocol error: unexpected TxRequest type")
```
Comment (therealssj) [inline src/skywallet/transaction.go]: TXINPUT?

File: src/skywallet/transaction.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+					} else {
+						msg, err = s.sendOutputs(index, len(s.Outputs)-index)
+						if err != nil {
+							return nil, err
+						}
+						s.state++
+						index = 0
+					}
+				} else {
+					return nil, fmt.Errorf("protocol error: unexpected TxRequest type")
```
Comment (therealssj) [inline src/skywallet/transaction.go]: TXOUTPUT?

File: src/skywallet/transaction.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+				return nil, err
+			}
+		default:
+			return nil, fmt.Errorf("unexpected response message type from hardware wallet")
+		}
+	}
+}
+
+func (s *SkycoinTransactionSigner) initSigningProcess() (wire.Message, error) {
+	return s.Device.SignTx(len(s.Outputs), len(s.Inputs), "Skycoin", s.Version, s.LockTime, "dkdji9e2oidhash")
```
Comment (therealssj) [inline src/skywallet/transaction.go:139]: what is this hardcoded value here?

**Review by therealssj — COMMENTED** [review]
looks good overall 
good work 👍

File: src/skywallet/transaction.go (hunk 1 above)
Comment (Midren) [reply src/skywallet/transaction.go]: Yeah, bad error message. Will change to `Expected TxInput, but got (TxRequest type, which came)`

File: src/skywallet/transaction.go (hunk 2 above)
Comment (Midren) [reply src/skywallet/transaction.go]: Will change to `Expected TxOutput, but got (TxRequest type, which came)`

File: src/skywallet/transaction.go (hunk 3 above)
Comment (Midren) [reply src/skywallet/transaction.go:139]: For long transaction support, the SignTx message was added, which initiate transaction signing with tx_hash, which is not used. As a function to make the message needed some tx_hash, some random tx was put. 

Maybe, a good idea would be just to pass empty string

Comment (therealssj) [reply src/skywallet/transaction.go:139]: ah okay.
would be nice to put it in the comments somewhere

**Pull request merged**